flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/FileCatalogStore.java

```diff
@@ -97,25 +114,29 @@ public void storeCatalog(String catalogName, CatalogDescriptor catalog)
             throws CatalogException {
         checkOpenState();
 
-        Path filePath = getCatalogPath(catalogName);
+        Path catalogPath = getCatalogPath(catalogName);
         try {
-            File file = filePath.toFile();
-            if (file.exists()) {
+            FileSystem fs = catalogPath.getFileSystem();
+
+            if (fs.exists(catalogPath)) {
                 throw new CatalogException(
                         String.format(
                                 "Catalog %s's store file %s is already exist.",
-                                catalogName, filePath));
+                                catalogName, catalogPath));
+            }
+
+            try (FSDataOutputStream os = fs.create(catalogPath, WriteMode.NO_OVERWRITE)) {
+                YAML_MAPPER.writeValue(os, catalog.getConfiguration().toMap());
             }
-            // create a new file
-            file.createNewFile();
-            String yamlString = yaml.dumpAsMap(catalog.getConfiguration().toMap());
-            FileUtils.writeFile(file, yamlString, charset);
-            LOG.info("Catalog {}'s configuration saved to file {}", catalogName, filePath);
-        } catch (Throwable e) {
+
+            LOG.info("Catalog {}'s configuration saved to file {}", catalogName, catalogPath);
+        } catch (CatalogException e) {
+            throw e;
+        } catch (Exception e) {
```

Comment:
Can we refine the different exceptions here to avoid the above process for `CatalogException`?

Reply:
I'm not sure what do you mean by refine. For example separating the operations that can throw not expected checked exception to multiple try/catch blocks to avoid catching `CatalogException` explicitly would only complicate the code more, I think. IMO this is not an anti-pattern at all, there are multiple examples in the codebase for it already, e.g. [here](https://github.com/apache/flink/blob/dfb9cb851dc1f0908ea6c3ce1230dd8ca2b48733/flink-filesystems/flink-hadoop-fs/src/main/java/org/apache/flink/runtime/fs/hdfs/HadoopFsFactory.java#L193).

Reply:
For example, we only need to catch IOException, StreamWriteException, DatabindException for the above code and convert them to CatalogException and rethrow CatalogException here

Reply:
Ah, first I thought that would make the logic behavior different because that will ignore any `RuntimeException`, which the general `Exception` will catch, but that can also be added so nvm. Will do this.

Reply:
Okay, I was mistaken, cause `CatalogException` extends `RuntimeException`, so `catch(IOException | RuntimeException e) ...` is would end up an unnecessary wrapping again. The viable alternative would be an `if` I think:
```java
try {
  // ...
} catch (Exception e) {
  if (e instanceof CatalogException) {
    throw e;
  }
  throw new CatalogException(...);
}
```
But personally I am leaning towards to my original solution, because it is more compact and IMO more readable as well.